src/lazy/binary/raw/v1_1/binary_buffer.rs

```diff
@@ -997,16 +994,22 @@ impl<'a> BinaryBuffer<'a> {
             return IonResult::incomplete("a length-prefixed e-expression", self.offset);
         }
         let matched_bytes = self.slice(0, total_length);
-        let macro_ref = self
+
+        // Get a reference to the macro that lives at that address
+        let macro_def = self
             .context
             .macro_table()
             .macro_at_address(macro_address)
             .ok_or_else(|| {
                 IonError::decoding_error(format!(
                     "invocation of macro at unknown address '{macro_address:?}'"
                 ))
-            })?
-            .reference();
+            })?;
+        let macro_ref = MacroRef::new(
+            QualifiedAddress::new(ModuleKind::Default, macro_address),
```

Comment:
🪧 `MacroRef`s now hold a `QualifiedAddress`, which is a `(module, address)` pair rather than just an `address`.